edge-util/src/Microsoft.Azure.Devices.Edge.Util/Proxy.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System;
+    using System.Net;
+    using Microsoft.Extensions.Logging;
+
+    public static class Proxy
+    {
+        public static Option<IWebProxy> Parse(string proxyUri, ILogger logger)
+        {
+            if (string.IsNullOrWhiteSpace(proxyUri))
+            {
+                return Option.None<IWebProxy>();
+            }
+
+            var uri = new Uri(proxyUri);
+            if (string.IsNullOrEmpty(uri.UserInfo))
+            {
+                // http://proxyserver:1234
+                logger.LogInformation($"Detected proxy {uri}");
+                return Option.Some<IWebProxy>(new WebProxy(uri));
+            }
+            else
+            {
+                string[] parts = uri.UserInfo.Split(':');
+                var credentials = new NetworkCredential
+                {
+                    UserName = Uri.UnescapeDataString(parts[0])
+                };
+
+                if (parts.Length == 1)
+                {
+                    // http://user@proxyserver:1234
+                    logger.LogInformation($"Detected proxy {uri}");
+                }
+                else
+                {
+                    // http://user:password@proxyserver:1234
+                    credentials.Password = Uri.UnescapeDataString(parts[1]);
+
+                    // log the proxy URI without the password
+                    int pos = uri.ToString().IndexOf(uri.UserInfo, StringComparison.InvariantCulture);
```

Comment:
Maybe just proxyUri?

Reply:
Just checking - InvariantCulture is intentional (instead of say Ordinal)?

Reply:
Are you suggesting I use the variable `proxyUri` instead of `uri.ToString()`? I did it this way for consistency, in case the Uri class does any kind of normalization on the input string (which was the `proxyUri` variable).

Since we're dealing with percent-encoded ASCII here, I think either will work. But I agree that Ordinal sounds like a better fit (just compare the raw numeric values of each code point). I'll switch to that, thanks.

Reply:
Using Ordinal now.